### ISSUE TITLE
[1.16.4] [Dev build] Crash & dupe related to the "Transportations" part of the mod

### ISSUE THREAD

**Issue by Creerio** [Chainmail-Studios/Astromine#360]
> ### General informations :

OS : Linux (Ubuntu 20.04.1 LTS)
Java version : 1.8.0_261 (not open-jdk)

Fabric version : 0.10.8 (latest)
Fabric API version : 0.25.7, same issue w/ the latest (0.26.1)

Mod version : Build from the developement-1.16.4 branch (Latest commit at the time : https://github.com/Mixinors/Astromine/tree/0fe1dd4e2cbe2d9fbce59f391f518244e7d2874e)
Other mods : NotEnoughCrashes to avoid a full crash, also tested w/ plenty of other ones.

>### Crash

### Summary of the issue

When right clicking on an Alternator, a Splitter, a Shredder or a Drain, the game crashes.

### How to reproduce it

Step 1 : Build the mod from the correct branch, add it in minecraft. Take any mod with it (or not), w/ fabric api of course and NotEnoughCrashes if you want to avoid crashing completely the game.
Step 2 : Go in a world, can be multiplayer (if you want to try since I didn't).
Step 3 : Put an Alternator, a Splitter, a Shredder or a Drain in the world and right click on it.
Step 4 : Done. You successfully crashed your game.

### Crash reports
W/ Fabric API 0.25.7 : [crash-2020-11-23_16.44.33-client.txt](https://github.com/Mixinors/Astromine/files/5584353/crash-2020-11-23_16.44.33-client.txt)
W/ Fabric API 0.26.1 : [crash-2020-11-23_17.31.29-client.txt](https://github.com/Mixinors/Astromine/files/5584354/crash-2020-11-23_17.31.29-client.txt)
Video : https://youtu.be/G9tbqtWdZbs?t=44

>### Dupe

### Summary of the issue

When using conveyors (any kind of it, not just basic, advanced, elite, vertical or downward) to move an item/block, it duplicates itself.

### How to reproduce it

Step 1 : Build the mod from the correct branch, add it in minecraft. Take any mod with it (or not), w/ fabric api of course and NotEnoughCrashes if you want to avoid crashing completely the game.
Step 2 : Go in a world, can be multiplayer (if you want to try since I didn't).
Step 3 : Put multiple conveyors in the world, they have to be next to each other and linked. It can be an input/output for an electric furnace for example.
Step 4 : Drop an item on one of the conveyors or make an item use your output conveyor.
Step 5: Well.. it's duping.

Video of the dupe : https://youtu.be/G9tbqtWdZbs

**Comment by Neubulae** [comment]
concen

**Comment by vini2003** [comment]
oh no

**Comment by Creerio** [comment]
I've build the latest version (https://github.com/Mixinors/Astromine/tree/8ec0e2c3b12f357048e6e098b0f8cc2523d6a8f6) to see if the issues are fixed, the dupe and crash issues are still here. One of the blocks however (the Drain) does no longer crash the game.

Since I needed this version to fix the cardinal issue w/ Cinderscape (#352 and #354), I've also went ahead and build a 1.16.3 version of the mod using the stable-1.16.3 branch and a modified version of the cardinal dependency in the gradle.properties file. Turns out it seems to work properly, so I'll use this version for now. 
It could be nice however if this fix was on curseforge. Some people might not know why the creative tab is a bit empty when using Cinderscape and Astromine. It could also help them wait for the next update.

I will still build the dev branch from time to time to see if the issues are fixed (unless you don't want to/close this issue). The new textures are really interesting so I hope it doesn't take too long to become a stable release.
